### DISCUSSION TITLE
Image quality/alignment causing digit ROI malformation?

### DISCUSSION THREAD

**Discussion by baudneo** [jomjol/AI-on-the-edge-device#2475] [Q&A]
I am uncertain how this is happening but, it is happening. I have been noticing that the '1' digit is not being recognized as a 1 but as a 0. When I check the image I can see that there is a misalignment of the 1 halfway through the digit. You can see the misalignment on another digit as well.

I did not notice this issue on the reference image so, I waited until about 20 seconds before the next round was going to start and I took a reference image to see if the '1' digit had the same malformation, it did not. The '1' digit seems to get malformed during either the image alignment or image capturing process. Please see attached screenshots.

Reference image 20 seconds before model input image is captured:
![Screenshot_2023-06-13_22-46-57](https://github.com/jomjol/AI-on-the-edge-device/assets/86508179/cd1d0cce-e4eb-4c0a-91c9-d11f40ab88f0)

Image the model runs on:
![Screenshot_2023-06-13_22-46-16](https://github.com/jomjol/AI-on-the-edge-device/assets/86508179/15f7b34f-d166-443b-a296-29263f7e019f)

You can clearly see about halfway through the digits that the image seems to shift. This is wreacking havoc on the results for the digit '1'. As an example, this is the RAW value obtained from that image: `05650.N`. 

I have the digit confidence set to `0.65` and I have experimented with using different image quality values, without success.

Has anyone else had this happen to them?

**Comment by baudneo** [comment]
It seems to be the alignment that is the issue. I caught the captured image before alignment and it does not have the layer 'shift' in the image until after the image is aligned.

Captured image before alignment:
![Screenshot_2023-06-13_23-16-54](https://github.com/jomjol/AI-on-the-edge-device/assets/86508179/071d977e-aa2b-4573-ad14-2cd5f6f19db1)


After alignment:
![Screenshot_2023-06-13_23-19-32](https://github.com/jomjol/AI-on-the-edge-device/assets/86508179/b7e9161a-0dea-497a-b2aa-f0dad8bd52c7)

**Comment by ghost** [comment]
It comes from the alignment process. It rotates the image. We using a very small device and the algorithm is not complex. 
You could rotate the mount to 0 rotation on the reference image. But in your case I would try to change the focus of the camera to have more sharpen images. The small shift should not have a big impact on the recognition.

**Reply by baudneo** [reply]
Nice, thank you for the advice. I'm trying to fine tune it as best I can.
